src/resolve/resolved-manager.c

```diff
@@ -196,6 +197,40 @@ static int manager_process_address(sd_netlink *rtnl, sd_netlink_message *mm, voi
         return 0;
 }
 
+static int manager_process_route(sd_netlink *rtnl, sd_netlink_message *mm, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        Link *l;
+        uint16_t type;
+        int r;
+
+        assert(rtnl);
+        assert(mm);
+
+        r = sd_netlink_message_get_type(mm, &type);
+        if (r < 0) {
+                log_warning_errno(r, "Failed not get message type, ignoring: %m");
+                return 0;
+        }
+
+        if (!IN_SET(type, RTM_NEWROUTE, RTM_DELROUTE)) {
+                log_warning("Unexpected message type %u when processing route, ignoring.", type);
+                return 0;
+        }
+
+        /* Routing table changes may affect the accessibility of DNS servers. */
+        HASHMAP_FOREACH(l, m->links) {
+                LIST_FOREACH(servers, s, l->dns_servers)
+                        dns_server_reset_accessible(s);
+        }
+
+        LIST_FOREACH(servers, s, m->dns_servers)
+                dns_server_reset_accessible(s);
```

Comment:
can you introduce a new helper call manager_reset_all_servers_accessible() or so that just cntains these two loops?